sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -143,16 +145,30 @@ public virtual void StartTestRecording()
                 throw new IgnoreException((string) test.Properties.Get("SkipRecordings"));
             }
             Recording = new TestRecording(Mode, GetSessionFilePath(), Sanitizer, Matcher);
+            ValidateClientInstrumentation = Recording.HasRequests;
         }
 
         [TearDown]
         public virtual void StopTestRecording()
         {
-            bool save = TestContext.CurrentContext.Result.Outcome.Status == TestStatus.Passed;
+            bool testPassed = TestContext.CurrentContext.Result.Outcome.Status == TestStatus.Passed;
+
+            if (ValidateClientInstrumentation && testPassed)
```

Comment:
I think this logic might be more clear if `ValidateClientInstrumentation` was changed to `IsClientInstrumented` and this line changes to:
```suggestion
            if (Recording.HasRequests && IsClientInstrumented && testPassed)
```

Reply:
Looking further down, I see we use this to set whether instrumentation should be validated. Perhaps just differentiating between "ShouldValidate" and "IsInstrumented" would be better then?

Reply:
You can't check `Recording.HasRequests` here as requests are removed from the recording during the playback. So `Recording.HasRequests` would always be false. That's why the logic looks a bit strange.